kong/cmd/utils/prefix_handler.lua

```diff
@@ -40,33 +42,48 @@ client:request { \
 ]]
 
 local resty_bin_name = "resty"
+local resty_version_pattern = "nginx[^\n]-openresty[^\n]-([%d%.]+)"
+local resty_compatible = version.set(meta._DEPENDENCIES.nginx_from, meta._DEPENDENCIES.nginx_to)
 local resty_search_paths = {
   "/usr/local/openresty/bin",
   ""
 }
 
+local function is_openresty(bin_path)
+  local cmd = fmt("%s -V", bin_path)
+  local ok, _, _, stderr = pl_utils.executeex(cmd)
+  log.debug("%s: '%s'", cmd, stderr:sub(1, -2))
+  if ok and stderr then
+    local version_match = stderr:match(resty_version_pattern)
+    if not version_match or not resty_compatible:matches(version_match) then
+      log.verbose("incompatible 'resty' found at %s. Kong requires version"..
+                  " %s, got %s", bin_path, tostring(resty_compatible),
+                  version_match)
+      return false
+    end
+    return true
+  end
+  log.debug("OpenResty 'resty' executable not found at %s", bin_path)
+end
```

Comment:
the resty cli is included in the openresty package, so version should never be changing without an openresty version change. That's why I used the openresty one. Also less maintenance...